src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java

```diff
@@ -248,13 +249,23 @@ public void removeGroupAndSubgroups(GroupNodeViewModel group) {
             //final UndoableAddOrRemoveGroup undo = new UndoableAddOrRemoveGroup(groupsRoot, node, UndoableAddOrRemoveGroup.REMOVE_NODE_AND_CHILDREN);
             //panel.getUndoManager().addEdit(undo);
 
+            removeGroupsAndSubGroupsFromEntries(group);
             group.getGroupNode().removeFromParent();
 
             dialogService.notify(Localization.lang("Removed group \"%0\" and its subgroups.", group.getDisplayName()));
             writeGroupChangesToMetaData();
         }
     }
 
+    private void removeGroupsAndSubGroupsFromEntries(GroupNodeViewModel group) {
+        for (GroupNodeViewModel child: group.getChildren()) {
+            removeGroupsAndSubGroupsFromEntries(child);
+        }
+
+        List<BibEntry> entriesInGroup = group.getGroupNode().getEntriesInGroup(this.currentDatabase.get().getEntries());
```

Comment:
You need to check if the `group` is an explicit group. Otherwise, removing a dynamic keyword-based group also removes the keyword from the entries that are matched by this group - this behavior is probably not what we want.

Reply:
Ok, thanks for clarifying. But then what do I do if it is a dynamic keyword group? No removal at all?

Reply:
Yes, I would just ignore every other type of group but the explicit group (or ask the user for confirmation about the removal).

Reply:
Ok, done!